tests/core/unit/middleware/data.tsx

```diff
@@ -0,0 +1,254 @@
+const { it, afterEach, beforeEach } = intern.getInterface('bdd');
+const { describe: jsdomDescribe } = intern.getPlugin('jsdom');
+const { assert } = intern.getPlugin('chai');
+import { sandbox } from 'sinon';
+import { renderer, tsx, create } from '../../../../src/core/vdom';
+import dataMiddleware, { createDataMiddleware, Resource, ResourceWrapper } from '../../../../src/core/middleware/data';
+import { createResolvers } from '../../support/util';
+
+const resolvers = createResolvers();
+
+const sb = sandbox.create();
+const invalidatorStub = sb.stub();
+
+let resourceStub = {
+	getOrRead: sb.stub(),
+	registerInvalidator: sb.stub()
+};
+
+jsdomDescribe('data middleware', () => {
+	beforeEach(() => {
+		resolvers.stub();
+	});
+	afterEach(() => {
+		resolvers.restore();
+		sb.resetHistory();
```

Comment:
do we use this abbreviation elsewhere? `sb`?

Reply:
believe so

Reply:
<img width="203" alt="Screenshot 2020-02-07 at 17 00 48" src="https://user-images.githubusercontent.com/814453/74049121-65016380-49cb-11ea-8dc7-bc9020a45751.png">
everywhere